examples/alicebob-grant-and-line-by-line-PRE.py

```diff
@@ -32,17 +35,43 @@
 ALICE.network_bootstrap([("localhost", 3601)])
 
 # Alice grants to Bob.
-policy = ALICE.grant(BOB, label, network_middleware, m=m, n=n,
+BOB = Bob()
+policy = ALICE.grant(BOB, label, m=m, n=n,
                      expiration=policy_end_datetime)
-hrac, treasure_map = policy.hrac(), policy.treasure_map
 
-# Bob can re-assemble the hrac himself with knowledge he already has.
-hrac = keccak_digest(bytes(ALICE.stamp) + bytes(BOB.stamp) + label)
-BOB.join_policy(ALICE, hrac, node_list=[("localhost", 3601)])
-
-# Now, Alice and Bob are ready for some throughput.
+# Alice puts her public key somewhere for Bob to find later...
+alices_pubkey_saved_for_posterity = bytes(ALICE.stamp)
```

Comment:
Is stamp starting to break down here?
We're admitting this is a public key, but not the type of public key. Stamp doesn't help make this more clear, imho.

Reply:
I'm not sure I understand what you mean by "not the type of public key".  `SignatureStamp` readily casts to bytes when concatenated.  In this case I just wanted to show that it is passed as arbitrary bytes hither and yon.

I mean, what might we do instead?  `bytes(ALICE.public_keys(SigningPower))`?  That's worse, no?

Reply:
It's more verbose, but ten-fold more explicit on which key it is. I don't think this is an either/or situation here. There should probably be a more explicit, yet simple method of doing this.

Reply:
I would agree that it took me a few seconds to understand that `stamp` is actually a public key. Although it's nice and short (and a verb!), it's not in line with common intuition

Reply:
Yeah, in isolation, I can definitely see how `bytes(ALICE.public_keys(SigningPower))` adds clarity.  But don't we *want* the user to pass Alice's stamp for these sorts of purposes?  It only looks strange here because it's a one-off in a demo.  But if they had a function:

```
def sign_something_and_return_signature_and_public_key(something, signer):
...
```

...then obviously we want to train them that Alice's `stamp` can be used as both a signer and a public key, right?

Maybe this isn't the right moment to teach that, in which case I'm fine switching to `bytes(ALICE.public_keys(SigningPower))`.  However, we might also be able to make it a good teachable moment with a quick additional comment.

Reply:
I'd lean more towards `bytes(ALICE.public_keys(SigningPower))` because it is not only avoids confusion with regard to the type of public key is at play here, it also introduces newbies to the general concept of powers (SigningPower, CrypoPower, etc) with a relatively straightforward example.

Also, we can always use a comment to explain what @jMyles is keen on teaching here – that Alice's `stamp` can be used as both a signer and a public key.